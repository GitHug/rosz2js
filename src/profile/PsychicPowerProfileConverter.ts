import AbstractProfileConverter from './AbstractProfileConverter';
import { PsychicPowerProfile, BSPsychicPowerCharacteristic, TypeName } from '../types';

class PsychicPowerProfileConverter extends AbstractProfileConverter<PsychicPowerProfile, BSPsychicPowerCharacteristic> {
  constructor() {
    super({
      typeName: TypeName.PSYCHIC_POWER,
      name: '-',
      warpCharge: '-',
      range: '-',
      details: '-'
    });
  }

  protected getProperty(bsCharacteristic: BSPsychicPowerCharacteristic): Partial<PsychicPowerProfile> {
    switch (bsCharacteristic.$.name) {
      case 'Warp Charge':
        return { warpCharge: bsCharacteristic._ };
      case 'Range':
        return { range: bsCharacteristic._ };
      case 'Details':
        return { details: bsCharacteristic._ };
    }
  }
}

export default PsychicPowerProfileConverter;
