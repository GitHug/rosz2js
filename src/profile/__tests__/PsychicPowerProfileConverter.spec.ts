import PsychicPowerProfileConverter from '../PsychicPowerProfileConverter';
import { BSTypeName, TypeName } from '../../Types';

describe('PsychicPowerProfileConverter', () => {
  it('should convert psychic power abilities', () => {
    const converter = new PsychicPowerProfileConverter();

    const profile = converter.convert({
      $: {
        typeName: BSTypeName.PSYCHIC_POWER,
        name: 'Kablam!'
      },
      characteristics: [
        {
          characteristic: [
            {
              $: {
                name: 'Warp Charge'
              },
              _: '99'
            },
            {
              $: {
                name: 'Range'
              },
              _: '25mi'
            },
            {
              $: {
                name: 'Details'
              },
              _: 'Forbidden lore'
            }
          ]
        }
      ]
    });

    expect(profile).toEqual({
      typeName: TypeName.PSYCHIC_POWER,
      name: 'Kablam!',
      warpCharge: '99',
      range: '25mi',
      details: 'Forbidden lore'
    });
    expect(profile).toMatchSnapshot();
  });

  it('should have default values', () => {
    const converter = new PsychicPowerProfileConverter();

    const profile = converter.convert({
      $: {
        typeName: BSTypeName.PSYCHIC_POWER,
        name: 'Kablam!'
      },
      characteristics: []
    });

    expect(profile).toEqual({
      typeName: TypeName.PSYCHIC_POWER,
      name: 'Kablam!',
      warpCharge: '-',
      range: '-',
      details: '-'
    });
    expect(profile).toMatchSnapshot();
  });
});
