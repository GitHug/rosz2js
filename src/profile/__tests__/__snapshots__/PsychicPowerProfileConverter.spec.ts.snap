// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PsychicPowerProfileConverter should convert psychic power abilities 1`] = `
Object {
  "details": "Forbidden lore",
  "name": "Kablam!",
  "range": "25mi",
  "typeName": "Psychic Power",
  "warpCharge": "99",
}
`;

exports[`PsychicPowerProfileConverter should have default values 1`] = `
Object {
  "details": "-",
  "name": "Kablam!",
  "range": "-",
  "typeName": "Psychic Power",
  "warpCharge": "-",
}
`;
